sandbox/cgroup/monitor_test.go

```diff
@@ -0,0 +1,247 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup_test
+
+import (
+	"os"
+	"path"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/sandbox/cgroup"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type monitorSuite struct {
+	testutil.BaseTest
+
+	tmp string
+	ch  chan string
+
+	inotifyWait time.Duration
+}
+
+var _ = Suite(&monitorSuite{})
+
+func (s *monitorSuite) SetUpTest(c *C) {
+	s.BaseTest.SetUpTest(c)
+
+	s.tmp = c.MkDir()
+	s.ch = make(chan string)
+	s.AddCleanup(func() { close(s.ch) })
```

Comment:
It's not very likely but if something goes wrong in the test it might happen that we close this channel and the goroutine running `watcherMainLoop` later writes to it, resulting in a panic. I think right now this can only happen if a test fails and we exit it before the watch is finished. It's related to this broader issue https://github.com/snapcore/snapd/pull/12280#r1043324293 of control over the channel and synchronization between writer/closer

Reply:
Yeah, this is a good observation and the fact that this can panic is a bit troublesome, I think this needs a closer look

Reply:
Sent a patch to fix this: https://github.com/snapcore/snapd/pull/12280/commits/c417e072800665882d95abb60636854ffcfd8405